apps/src/util/globalEdition.ts

```diff
@@ -0,0 +1,13 @@
+import {get} from 'js-cookie';
+
+export const getGlobalEditionRegion = () => {
+  const geRegionScript = document.querySelector(
+    'script[data-ge-region]'
+  ) as HTMLScriptElement;
+
+  const geRegion = geRegionScript
+    ? geRegionScript.dataset.geRegion
+    : get('ge_region');
+
+  return geRegion || null;
```

Comment:
I solved this a different way via https://github.com/code-dot-org/code-dot-org/pull/61471/files#diff-d475179c69f86dc870de373d45bcf9c44d22611e55a41e697e08de9c0ba52f37R24

```
window.location.pathname.match(/^\/global\/(fa)/)?.[1] || 'root';
```

Curious if there is a good reason I'm not considering to have this type of complexity, here, instead?

Reply:
We need to use the cookie value to ensure consistency between the backend and frontend

Reply:
The less hardcoded and duplicated business logic we have, the easier it will be to apply changes or perform refactoring

Reply:
So, it's more about the region the User is currently locked to rather than the represented region of the current page. That makes sense.

Though, I'm curious: what is a possible scenario where the backend and frontend are inconsistent if you are just looking at the current URL?

Reply:
For example, we don’t redirect AJAX requests from React, so the only way for the backend to identify if the request is from the Global Edition region is through cookies. Additionally, currently, Pegasus Farsi pages are simply static pages and do not follow the Global Edition business logic that Studio does